node/src/store.rs

```diff
@@ -1,14 +1,361 @@
+use async_trait::async_trait;
+use celestia_types::ExtendedHeader;
+use core::fmt::Debug;
+use dashmap::mapref::entry::Entry;
+use dashmap::DashMap;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+use tendermint::Hash;
+use thiserror::Error;
+use tracing::{error, info, instrument};
+
+type Result<T, E = StoreError> = std::result::Result<T, E>;
+
+pub type WrappedStore = Arc<Box<dyn Store + Sync + Send>>;
```

Comment:
Usually this pattern has `Boxed` as suffix and without the `Arc`.

```suggestion
pub type StoreBoxed = Box<dyn Store + Sync + Send>;
```

Reply:
renamed :+1: